ping-protect/src/androidTest/java/org/forgerock/android/auth/PingOneProtectEvaluateCallbackTest.kt

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2024 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+package org.forgerock.android.auth
+
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.forgerock.android.auth.callback.TextOutputCallback
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Assert
+import org.junit.FixMethodOrder
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.MethodSorters
+
+@RunWith(AndroidJUnit4::class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+class PingOneProtectEvaluateCallbackTest : BasePingOneProtectTest() {
+    private val TREE = "TEST_PING_ONE_PROTECT_EVALUATE"
+
+    @Test
+    fun test01EvaluateNoInit() {
+        var evaluateFailure = false
+        val nodeListenerFuture: PingOneProtectNodeListener = object : PingOneProtectNodeListener(
+            context, "evaluate-no-init"
+        ) {
+            val nodeListener: NodeListener<FRSession?> = this
+            override fun onCallbackReceived(node: Node) {
+
+                if (node.getCallback(PingOneProtectEvaluationCallback::class.java) != null) {
```

Comment:
i think we can simplify this 

`   override fun onCallbackReceived(node: Node) {
                
                node.getCallback(PingOneProtectEvaluationCallback::class.java)?.let {
                    val scope = CoroutineScope(Dispatchers.Default)
                    scope.launch {
                        try {
                            it.getData(context)
                        }
                        catch (e: Exception) {
                            evaluateFailure = true
                            assertThat("PingOneSignals SDK is not initialized", e.message === "PingOneSignals SDK is not initialized")
                        }
                        node.next(context, nodeListener)
                    }
                    return
                }

                node.getCallback(TextOutputCallback::class.java)?.let {
                    val callback = node.getCallback(
                        TextOutputCallback::class.java
                    )

                    assertThat("The Protect Evaluation node should trigger the Client Error outcome", callback.message == "Client Error")
                    node.next(context, nodeListener)
                    return
                }
                
                super.onCallbackReceived(node)
            }`

Reply:
Thanks! I applied the suggestion to all test cases...